### ISSUE TITLE
Dependency issues in GHC 7.10

### ISSUE THREAD

**Issue by matthewleon** [jprupp/haskoin-core#168]
GHC 7.10 isn't supported by LTS Haskell, so I try to use a Stackage nightly from `https://www.stackage.org/nightly/cabal.config`

Of course, I then fall into that altogether unpleasant place known as Cabal hell:

```
mlg-ell:haskoin mattleon$ cabal install
Resolving dependencies...
cabal: Could not resolve dependencies:
trying: haskoin-0.2.0 (user goal)
next goal: time (dependency of haskoin-0.2.0)
rejecting: time-1.5.0.1/installed-30e... (conflict: haskoin => time>=1.4 &&
<1.5)
rejecting: time-1.5.0.1, 1.5, 1.4.2, 1.4.1, 1.4.0.2, 1.4.0.1, 1.4, 1.3,
1.2.0.5, 1.2.0.4, 1.2.0.3, 1.2.0.2, 1.2.0.1, 1.2, 1.1.4, 1.1.3, 1.1.2.4,
1.1.2.3, 1.1.2.2, 1.1.2.1, 1.1.2.0, 1.0 (global constraint requires installed
instance)
Dependency tree exhaustively searched.
```

For giggles, I tried `--allow-newer` as well. It gets surprisingly far and then chokes:

```
[ 4 of 44] Compiling Network.Haskoin.Script.Types ( Network/Haskoin/Script/Types.hs, dist/build/Network/Haskoin/Script/Types.o )

Network/Haskoin/Script/Types.hs:77:10:
    No instance for (GHC.Generics.Generic PushDataType)
      arising from a use of ‘Control.DeepSeq.$gdmrnf’
    In the expression: Control.DeepSeq.$gdmrnf
    In an equation for ‘rnf’: rnf = Control.DeepSeq.$gdmrnf
    In the instance declaration for ‘NFData PushDataType’
cabal: Error: some packages failed to install:
haskoin-0.2.0 failed during the building phase. The exception was:
ExitFailure 1
```

**Comment by matthewleon** [comment]
A suggestion: get this working with stack. https://github.com/commercialhaskell/stack

**Comment by jprupp** [comment]
Hello, yes, Stack is coming to Haskoin. I will also fix GHC 7.10 compatibility issues, as long as they don't break the package for GHC 7.8.

**Comment by jprupp** [comment]
We have moved to Stack and we are supporting LTS Haskell 2.15 now. When GHC 7.10 becomes the default compiler for LTS Haskell, we will add support.
